### ISSUE TITLE
[ENHANCEMENT]

### ISSUE THREAD

**Issue by AnkithThodupunoori** [Airspace-Encounter-Models/em-core#15]
## Instructions

Read the instructions and fill out your responses at the bottom.

### Relation

Is your feature request related to a problem? Please describe. Please provide a clear and concise description of what the problem is. Ex. I'm always frustrated when [...]

### Relation
.hdr files which are located in DEM_GLOBE folder are unable to be read by code. when ever i execute RUN_Airspace_1, it throws an error saying "e10g" file format  unable to read. i have tried to with automatic generated file format by running setup script and with manual downloaded file. both doesnt work.
### Description
the error thrown by matlab looks like shown bellow. can i know how to get rid of this issue and execute the code further.

Error using georasterinfo (line 70)
Unable to read 'e10g'. Format may not be supported, file may be corrupt, or a supporting file may have been
specified.

Error in msl2agl (line 189)
        info = georasterinfo(inFile);

Error in readAirspace (line 108)
parfor i=1:1:numel(LAT_deg)

Error in RUN_Airspace_1 (line 30)
airspaceUS = readAirspace('inFile',inFile,...

**Comment by aweinert-MIT** [comment]
What version of MATLAB are you using? Can you please confirm you successfully ran the startup scripts?

**Comment by AnkithThodupunoori** [comment]
my matlab version is R2021b, i executed the setup file, but i am not sure it executed completely, i attached a picture what it is showing at the end when i run setup file.

![Em_Core issues](https://user-images.githubusercontent.com/84036542/189992149-b587bf95-3523-4766-baeb-5c181e91978e.png)

**Comment by aweinert-MIT** [comment]
`Unable to establish SSL connection`, so it looks like the files were not downloaded. You can confirm this by inspecting if the files are in the [`em-core/data/DEM-GLOBE](https://github.com/Airspace-Encounter-Models/em-core/tree/master/data/DEM-GLOBE) directory.

A common culprit is not setting a web proxy if you're working on a managed network.

**Comment by AnkithThodupunoori** [comment]
files are already in dem_gobe folder
![dem_globe files](https://user-images.githubusercontent.com/84036542/189996980-4fcb2577-8a78-4a97-b2a0-93ccc0fb59d3.png)

**Comment by aweinert-MIT** [comment]
That's good. The files are there, so we can assume the system environment variable was set correctly. Is `em-core` added to your MATLAB path?

**Comment by AnkithThodupunoori** [comment]
yes it is added to the path

**Comment by aweinert-MIT** [comment]
`georasterinfo` is part of the MATLAB mapping toolbox. Can you please confirm you have access to the mapping toolbox?

**Comment by AnkithThodupunoori** [comment]
yes, mapping tool box is installed in my matlab.
![image](https://user-images.githubusercontent.com/84036542/189998430-afc065de-cfe4-4cda-bcf3-1a5a329a73b0.png)

**Comment by AnkithThodupunoori** [comment]
in **gerasterinfo** , at line 37, the command is not able to execute, thats the one throwing error
![image](https://user-images.githubusercontent.com/84036542/189998885-d93809cd-033f-4785-9cb8-59603b12f61f.png)

**Comment by AnkithThodupunoori** [comment]
is it a file format issue or the matlab file "georasterinfo"  ?

**Comment by aweinert-MIT** [comment]
On Line 108, replace `parfor` with `for`, so it reads `fori=1:1:numel(LAT_deg)`

**Comment by aweinert-MIT** [comment]
> On Line 108, replace `parfor` with `for`, so it reads `fori=1:1:numel(LAT_deg)`
> 
> 
> 
> 
> 
> 
> 
> 

@AnkithThodupunoori Where you able to test this proposed change? Do you have access to the parallel toolbox. If no, replace 'parfor' may address the bug. Also replacing it should produce a different error message, which hopefully should be more insightful.

**Comment by AnkithThodupunoori** [comment]
i have the parfor tool box. anyway i made the changes as you asked to replace the parfor to for loop,  this is the error i am getting after executing the code.
![image](https://user-images.githubusercontent.com/84036542/191283898-f1de35e5-d88b-47c0-812f-5e83902652a3.png)

**Comment by aweinert-MIT** [comment]
what is `inFile` on line 189? That error reads like it can find the `e10g` file, but you're previously shown it is downloaded:

> files are already in dem_gobe folder ![dem_globe files](https://user-images.githubusercontent.com/84036542/189996980-4fcb2577-8a78-4a97-b2a0-93ccc0fb59d3.png)

**Comment by AnkithThodupunoori** [comment]
infile represents the "e10" file, matlab is able to find the file but it just cant read the file.
![image](https://user-images.githubusercontent.com/84036542/191791692-c17cda2c-ef70-42b8-be6a-f672116d3c0a.png)

**Comment by aweinert-MIT** [comment]
If you change directory to `DEM-GLOBE`, can you load the file via command line?

**Comment by AnkithThodupunoori** [comment]
it gives the same error, saying unable to read the file format or file may be corrupted, as shown below.
![image](https://user-images.githubusercontent.com/84036542/191807524-8bf4f1e4-657e-4a46-80a0-1753274366aa.png)

**Comment by AnkithThodupunoori** [comment]
if i use the fileread function, it reads the data in characters but cannot be displayed as the legth of the character is too large.
![image](https://user-images.githubusercontent.com/84036542/191808651-1f20124c-4947-41be-b616-710a2183797c.png)

but when i use the same function which is used in georasterinfo, it throws the error as usual.
![image](https://user-images.githubusercontent.com/84036542/191808845-9d745576-f6ac-4177-9cf9-248c99a62b96.png)

**Comment by aweinert-MIT** [comment]
Have you tried redownloading `e10g` manually and confirm it isn't corrupted?

**Comment by AnkithThodupunoori** [comment]
Yes, Just now downloaded manually and tried to execute the code, it shows same error.

**Comment by aweinert-MIT** [comment]
The output should look something like this (_I slightly censored the `Filename` property_):
```matlab
>> inFile = [getenv('AEM_DIR_CORE') filesep 'data' filesep 'DEM-GLOBE' filesep 'e10g'];
>> info = georasterinfo(inFile)
Warning: Projection information missing from 'e10g'. Assuming the RasterReference property is a map raster reference object. 
> In georasterinfo>constructRasterReference (line 163)
In georasterinfo (line 52) 
info = 
  RasterInfo with properties:

                     Filename: ["\em-core\data\DEM-GLOBE\e10g"    "\em-core\data\DEM-GLOBE\e10g.hdr"]
             FileModifiedDate: [2004-05-07 13:31:30    2004-05-07 13:48:38]
                     FileSize: [129600000 340]
                   FileFormat: "Esri GridFloat"
                   RasterSize: [6000 10800]
                     NumBands: 1
                 NativeFormat: "int16"
         MissingDataIndicator: -500
                   Categories: []
                    ColorType: "unknown"
                     Colormap: []
              RasterReference: [1×1 map.rasterref.MapCellsReference]
    CoordinateReferenceSystem: []
                     Metadata: []
>> 
```

**Comment by aweinert-MIT** [comment]
Is both `e10g` and `e10g.hdr` in your directory?

**Comment by AnkithThodupunoori** [comment]
NO, I have only one file "e10g" in my directory.

**Comment by AnkithThodupunoori** [comment]
i tried to generate the output of the file using the code you posted above, this is how it showing.
![image](https://user-images.githubusercontent.com/84036542/192593417-d60e9ed6-2bd6-4101-b6d8-675610aebb78.png)

**Comment by aweinert-MIT** [comment]
The bug looks to be with using `georasterinfo` and not the MATLAB code in `em-core`. Only the `e10g` file is required and extracted from the zip downloaded using the `setup.sh` script. The file size of `e10g` is the same as what you reported and what I redownloaded this afternoon. I successfully tested `georasterinfo` using MATLAB R2020b and R2022a.

Brainstorming here, can you confirm `exist()` returns `2` and `which` returns something the mapping toolbox directory?

```matlab
>>  inFile = [getenv('AEM_DIR_CORE') filesep 'data' filesep 'DEM-GLOBE' filesep 'e10g'];
>> exist(inFile)
ans =
     2
>> which inFile
```

**Comment by AnkithThodupunoori** [comment]
This is what it return when used the suggested commands
![image](https://user-images.githubusercontent.com/84036542/192617051-11996b93-e9de-466a-9ea3-27cfb6dd6998.png)

**Comment by aweinert-MIT** [comment]
What about `which georasterinfo`?

**Comment by AnkithThodupunoori** [comment]
![image](https://user-images.githubusercontent.com/84036542/192618556-281fad71-4ddf-41a4-8a54-811b3d156f14.png)

**Comment by aweinert-MIT** [comment]
All those returns are what is expected

**Comment by AnkithThodupunoori** [comment]
So do you what exactly the problem here? is it the matlab version i am using?

**Comment by aweinert-MIT** [comment]
I don't think it is the MATLAB version because I tested a version older and newer than what you're using. MATLAB recognizes the file exists and `which` indicates the correct `georasterinfo` is on the path.

I'm a bit stumped on the bug. For sanity, can you try loading the file using [`readgeoraster`](https://www.mathworks.com/help/map/ref/readgeoraster.html)? It should fail like when using `georasterinfo`

**Comment by AnkithThodupunoori** [comment]
yes, looks like it failed same like before
![image](https://user-images.githubusercontent.com/84036542/192625241-a2382e96-a4c5-4dad-8504-690be688dba1.png)

**Comment by aweinert-MIT** [comment]
I'm still musing about what the bug could be here because its with a built-in MATLAB function.
